File: app/src/main/java/org/ole/planet/myplanet/ui/references/ReferenceFragment.java
package org.ole.planet.myplanet.ui.references;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import org.ole.planet.myplanet.R;
import org.ole.planet.myplanet.callback.OnHomeItemClickListener;
import org.ole.planet.myplanet.databinding.FragmentReferenceBinding;
import org.ole.planet.myplanet.databinding.RowReferenceBinding;
import org.ole.planet.myplanet.model.Reference;
import org.ole.planet.myplanet.ui.dictionary.DictionaryActivity;
import org.ole.planet.myplanet.ui.map.OfflineMapActivity;

import java.util.ArrayList;
import java.util.List;

public class ReferenceFragment extends Fragment {
    private FragmentReferenceBinding fragmentReferenceBinding;
    private RowReferenceBinding rowReferenceBinding;
    OnHomeItemClickListener homeItemClickListener;

    public ReferenceFragment() {
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        if (context instanceof OnHomeItemClickListener)
            homeItemClickListener = (OnHomeItemClickListener) context;
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        fragmentReferenceBinding = FragmentReferenceBinding.inflate(inflater, container, false);
        List<Reference> list = new ArrayList<>();
        list.add(new Reference(getString(R.string.maps), android.R.drawable.ic_dialog_map));
        list.add(new Reference(getString(R.string.engilsh_dictionary), R.drawable.ic_dictionary));
        fragmentReferenceBinding.rvReferences.setLayoutManager(new GridLayoutManager(getActivity(), 3));
        setRecyclerAdapter(list);
        return fragmentReferenceBinding.getRoot();
    }

    private void setRecyclerAdapter(List<Reference> list) {
        fragmentReferenceBinding.rvReferences.setAdapter(new RecyclerView.Adapter() {
            @NonNull
            @Override
            public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
                rowReferenceBinding = RowReferenceBinding.inflate(LayoutInflater.from(parent.getContext()), parent, false);
                return new ViewHolderReference(rowReferenceBinding.getRoot());
            }

            @Override
            public void onBindViewHolder(@NonNull RecyclerView.ViewHolder holder, int position) {
                rowReferenceBinding.title.setText(list.get(position).getTitle());
                rowReferenceBinding.icon.setImageResource(list.get(position).getIcon());
                rowReferenceBinding.getRoot().setOnClickListener(view -> {
                    if (position == 0)
                        startActivity(new Intent(getActivity(), OfflineMapActivity.class));
                    else {
                        startActivity(new Intent(getActivity(), DictionaryActivity.class));
                    }
                });
            }

            @Override
            public int getItemCount() {
                return list.size();
            }
        });
    }

    static class ViewHolderReference extends RecyclerView.ViewHolder {

        public ViewHolderReference(View itemView) {
            super(itemView);
        }
    }
}

File: app/src/main/java/org/ole/planet/myplanet/ui/references/ReferenceFragment.kt
package org.ole.planet.myplanet.ui.references

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.RecyclerView
import org.ole.planet.myplanet.R
import org.ole.planet.myplanet.callback.OnHomeItemClickListener
import org.ole.planet.myplanet.databinding.FragmentReferenceBinding
import org.ole.planet.myplanet.databinding.RowReferenceBinding
import org.ole.planet.myplanet.model.Reference
import org.ole.planet.myplanet.ui.dictionary.DictionaryActivity
import org.ole.planet.myplanet.ui.map.OfflineMapActivity

class ReferenceFragment : Fragment() {
    private var _binding: FragmentReferenceBinding? = null
    private val binding get() = _binding!!
    private lateinit var rowReferenceBinding: RowReferenceBinding
    private var homeItemClickListener: OnHomeItemClickListener? = null

    override fun onAttach(context: Context) {
        super.onAttach(context)
        homeItemClickListener = context as? OnHomeItemClickListener
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
        _binding = FragmentReferenceBinding.inflate(inflater, container, false)
        val list = listOf(
            Reference(getString(R.string.maps), android.R.drawable.ic_dialog_map),
            Reference(getString(R.string.english_dictionary), R.drawable.ic_dictionary)
        )
        binding.rvReferences.layoutManager = GridLayoutManager(activity, 3)
        setRecyclerAdapter(list)
        return binding.root
    }

    private fun setRecyclerAdapter(list: List<Reference>) {
        binding.rvReferences.adapter = object : RecyclerView.Adapter<ViewHolderReference>() {
            override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolderReference {
                rowReferenceBinding = RowReferenceBinding.inflate(LayoutInflater.from(parent.context), parent, false)
                return ViewHolderReference(rowReferenceBinding.root)
            }

            override fun onBindViewHolder(holder: ViewHolderReference, position: Int) {
                rowReferenceBinding.title.text = list[position].title
                rowReferenceBinding.icon.setImageResource(list[position].icon)
                rowReferenceBinding.root.setOnClickListener {
                    if (position == 0)
                        startActivity(Intent(activity, OfflineMapActivity::class.java))
                    else {
                        startActivity(Intent(activity, DictionaryActivity::class.java))
                    }
                }
            }

            override fun getItemCount(): Int = list.size
        }
    }

    class ViewHolderReference(itemView: View) : RecyclerView.ViewHolder(itemView)

    override fun onDestroyView() {
        _binding = null
        super.onDestroyView()
    }
}
